From 8b347fd922e4c19a3a0fb4873edbc8e30ca146ef Mon Sep 17 00:00:00 2001
From: marionm <mathis.marion@grenoble-inp.org>
Date: Fri, 20 Aug 2021 11:35:22 +0200
Subject: [PATCH 1/2] Kernel 5.2 compatibility for i2c_new_device

`i2c_new_device` disappeared in kernel 5.2 and onwards.
It is now required to use `i2c_new_client_device`, which
works the same way except for error checking.

Signed-off-by: Gwenhael GOAVEC <gwenhael.goavec@femto-st.fr>
Signed-off-by: marionm <mathis.marion@grenoble-inp.org>
---
 software/i2c-ocores/drivers/i2c/busses/i2c-ocores.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/software/i2c-ocores/drivers/i2c/busses/i2c-ocores.c b/software/i2c-ocores/drivers/i2c/busses/i2c-ocores.c
index b7a458d..518ede1 100644
--- a/software/i2c-ocores/drivers/i2c/busses/i2c-ocores.c
+++ b/software/i2c-ocores/drivers/i2c/busses/i2c-ocores.c
@@ -895,7 +895,11 @@ static int ocores_i2c_probe(struct platform_device *pdev)
 	/* add in known devices to the bus */
 	if (pdata) {
 		for (i = 0; i < pdata->num_devices; i++)
+#if KERNEL_VERSION(5, 2, 0) <= LINUX_VERSION_CODE
+			i2c_new_client_device(&i2c->adap, pdata->devices + i);
+#else
 			i2c_new_device(&i2c->adap, pdata->devices + i);
+#endif
 	}
 
 	return 0;
-- 
2.30.2

